Rename_by_time/main.py

```diff
@@ -0,0 +1,43 @@
+import os
+from math import log10
+
+files=os.listdir(".")
+
+files.sort(key=lambda x: os.path.getmtime(x))
+# print(files) # list index 0 hold the oldest file
+digits=int(log10(len(files)))+1
+
+print(f"\nPlease verify the new file names for accuracy: press 'p' to review the file names, or 'y'/'Y' to proceed with the renaming process. Press any other key to exit the program.\n")
+n=input()
+
+if(n=="y" or n=="Y" or n=="p"):
+    rename,hidden=0,0
+    for i in range(len(files)-1):
+        if(files[i][0]=="."):
+            hidden+=1
+            continue
+
+        if "_" in files[i] and files[i][0]!="_":
+            index=files[i].find("_")
+            isNum=files[index-1]
+        
+            if(isNum >="0" and isNum<="9"):
+                oldName=files[i][index+1:]
+            else:
+                oldName=files[i]
+        else:
+            oldName=files[i]
+
+        rename+=1
+        newfile_name=f"{str(rename).zfill(digits)}_{oldName}"
+
+        if(n=="p"):
+            print(f"{oldName} --> {newfile_name}")
+        else:
+            os.rename(oldName,newfile_name)
```

Comment:
Before rename, check if new file name is already exist or not to avoid exception

Reply:
Hello @JRudransh , the code is written in that way if the new file name is already present for example 01_filename before the code run, it automatically change the filename before renaming. For references, pls check this screenshot I have attached. 

### Before adding the new file name I am checking what can be the new file name could be
![Screenshot from 2023-07-22 09-58-30](https://github.com/avinashkranjan/Amazing-Python-Scripts/assets/92109154/0655e724-3a22-4f2d-9858-a2744fadb906)


### As you can check 01_vid1.c is one of the file name , so I make it already present in the directory and run the code. As you can check in preview, how it manage the existing same new file name by making it 13_01_filename

![image](https://github.com/avinashkranjan/Amazing-Python-Scripts/assets/92109154/a6f0b469-d462-45ed-84e3-7717551926b4)